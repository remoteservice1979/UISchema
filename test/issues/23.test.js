import React from "react";
import Form from "react-jsonschema-form";
import applyPagination from "../../src";
import renderer from "react-test-renderer";

let schema = {
  type: "object",
  properties: {
    firstName: {
      type: "string",
      default: "",
    },
    lastName: { type: "string" },
  },
};
let uiSchema = {
  "ui:order": ["firstName", "lastName"],
  firstName: {
    classNames: "col-md-12",
  },
};
let formData = {
  firstName: "Bill",
};

test("Initial formData rendered", () => {
  let ResForm = applyPagination(Form);
  const component = renderer.create(
    <ResForm schema={schema} uiSchema={uiSchema} formData={formData} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
