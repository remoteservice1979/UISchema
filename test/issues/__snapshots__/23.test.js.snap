// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial formData rendered 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group field field-object"
  >
    <fieldset>
      <div
        className="form-group field field-string  col-md-12"
      >
        <label
          className="control-label"
          htmlFor="root_firstName"
        >
          firstName
        </label>
        <div />
        <input
          autoFocus={false}
          className="form-control"
          disabled={false}
          id="root_firstName"
          label="firstName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value="Bill"
        />
        <div />
        <div />
      </div>
      <div
        className="form-group field field-string"
      >
        <label
          className="control-label"
          htmlFor="root_lastName"
        >
          lastName
        </label>
        <div />
        <input
          autoFocus={false}
          className="form-control"
          disabled={false}
          id="root_lastName"
          label="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
        <div />
        <div />
      </div>
    </fieldset>
    <div />
    <div />
  </div>
  <p>
    <button
      className="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </p>
</form>
`;
